src/Dns/DnsIpAddress.cs

```diff
@@ -102,10 +102,10 @@ public DnsIpAddress GetFlipped()
         {
             // rotate octets
             var flippedAddress =
-                ((address << 24) & 0xFF_00_00_00) |
-                ((address << 16) & 0x00_FF_00_00) |
```

Comment:
Good catch! The shifting of 16 should be 8. Your solution is fine in this instance because we are operating on an unsigned value (logical bitwise shifting is used).

I do tend to be in the habit of shifting first, then masking due to the behavior of arithmetic bitwise shifting (used for signed values). For example:
_Unsigned integer:_ `(0xFF_00_00_00 >> 8) == 0x00_FF_00_00 == 16,711,680`
_Signed integer:_ `(0xFF_00_00_00 >> 8) == 0xFF_FF_00_00 == -65,536`